### ISSUE TITLE
Row heigths can't be smaller than row header's text height.

### ISSUE THREAD

**Issue by KodjoSuprem** [handsontable/handsontable#4021]
Row heigths can't be smaller than rowheader text's height.

Both with Manual row resize plugin and HT rowHeights option.

Rows header content should be overlapped if needed.

Exemple on Excel:
![image](https://cloud.githubusercontent.com/assets/4376260/22296921/f8309ece-e31b-11e6-8d5c-40fac4fb062f.png)

**Comment by AMBudnik** [comment]
Hi @KodjoSuprem 
How you would like it to look like? Like hidden with an extra second border?

**Comment by KodjoSuprem** [comment]
It is like their is a minimum row size that cannot be overriden, with the rowHeights options.

Take a look at this exemple : http://jsfiddle.net/f75hboxj/
Row n°2 should be extra thin.
But it keeps a minimum size

**Comment by AMBudnik** [comment]
You're right. Even when there are no headers like here: http://jsfiddle.net/handsoncode/1uzs1jj9/ you still will get min height of 22px for a row. And the smallest value you can set is 4px http://jsfiddle.net/handsoncode/9p1nmjdp/
Currently Handsontable does not allow to cut text vertically.

**Comment by KodjoSuprem** [comment]
Ok so can you propose as workaround?
Currently I remove the row header innerHTML rows smaller than 20 px just like you did in your exemple.

But it is not the best solution.

**Comment by AMBudnik** [comment]
So basically you would like this row to be hidden (not deleted)? Can you create a draft (can be a drawing) with a desired look and a index structure that you would like to get? I guess that there's something we can do.

**Comment by KodjoSuprem** [comment]
Basically the minimum row height of 22px is critical.
Tables with headers should accept smallers height 

Exemple:
MS excel allow small row heights, header and row text are half visible (rows overlap)

![image](https://cloud.githubusercontent.com/assets/4376260/22590284/bd430334-ea0e-11e6-8c35-358b72cad13e.png)

For legal reasons document layouts have to be very specific so row size matter a lot for our clients. A lot of official reports use empty lines as separators. Those lines have a height smaller than 22px.

**Comment by AMBudnik** [comment]
Hi @KodjoSuprem 

thank you for additional information.

I tried to find a workaround however the `beforeRowResize` works exactly as `beforeColumnResize` with the reported bug: https://github.com/handsontable/handsontable/issues/3328

I will mark this issue as a request for a change.

**Comment by balmychan** [comment]
👍

**Comment by AMBudnik** [comment]
We should discuss this report. From one hand we have decided that the cell should have at least 23px pixels but on another, we try to keep compatibility with users' habits that are mainly created by Excel or Google Sheets.

**Comment by AMBudnik** [comment]
We should also think about doing the same operation for headers https://jsfiddle.net/joshenglish/3ec1tq3z/ (`A` is a bit long but we can't make it shorter)

**Comment by warpech** [comment]
Related: https://github.com/handsontable/handsontable/issues/6149

----

There is one additonal quirk. If you provide configuration for `rowHeights`, where all values are smaller than `24` (e.g. ` rowHeights: [10, 10, 23],`), the rows header is completely misaligned: 

https://jsfiddle.net/6ubL34mw/2/

![image](https://user-images.githubusercontent.com/566463/68773554-10e51e00-062c-11ea-9e70-ae50fd4b066b.png)

However, when at least one of the values equals `24` or higher, the misalignment does not happen. Values smaller than the row header's text height are simply ignored:

https://jsfiddle.net/6ubL34mw/1/

![image](https://user-images.githubusercontent.com/566463/68773528-03c82f00-062c-11ea-933a-a8764c261974.png)

**Comment by budnix** [comment]
Hi @KodjoSuprem, thanks for the report, and sorry it sat here for so long.

I tested this on the current version and it still behaves the way you described. `rowHeights` works as a minimum, so a row never becomes shorter than the tallest thing in it, and the row header's text is one of the things that holds it open. Dragging a row with Manual row resize hits the same limit.

I have a fix on a branch now. It makes `rowHeights` set the exact height of a row and clip whatever does not fit, which is the Excel behavior in your screenshot. It covers the row header too, so a short row no longer stays open just because of the header text, and a row you drag with Manual row resize stays at the size you drop it. `minRowHeights` keeps the current meaning, so if you want the value to stay a floor that content can grow past, you use that instead.

It will go out in an upcoming release.

**Comment by KodjoSuprem** [comment]
Damn, 10 years already?! Time flies... I remember I managed to hack a workaround and make it work as we wanted.
Good job. Thank you @budnix
